goworker.go

```diff
@@ -87,12 +90,71 @@ func Init() error {
 // while they wait for an available connection. Expect this
 // API to change drastically.
 func GetConn() (*RedisConn, error) {
+	if workerSettings.RedisSettings.MasterName == "" {
+		return getConn()
+	} else {
+		return getConnSentinel()
+	}
+}
+
+func getConn() (*RedisConn, error) {
 	resource, err := pool.Get(ctx)
 
 	if err != nil {
 		return nil, err
 	}
 	return resource.(*RedisConn), nil
+
+}
+func getConnSentinel() (*RedisConn, error) {
+	deadConnection := errors.New("Dead connection")
+	slaveConnection := errors.New("Stale connection (to slave, not master)")
+	try := func() (*RedisConn, error) {
+		resource, err := pool.Get(ctx)
+
+		if err != nil {
+			return nil, err
+		}
+		conn := resource.(*RedisConn)
+
+		// close the connection and remove it from the pool so that new
+		// connections get created
+
+		// if the instance does not ping back
+		_, err = conn.Do("ping")
+		if err != nil {
+			conn.Close()
+			PutConn(nil)
+			return nil, deadConnection
+		}
+
+		// or if the instance is not a master anymore
+		role, err := redis.String(conn.Do("role"))
```

Comment:
(sorry for the delayed getting to this)

If we execute `role` on a non-sentinel setup, it will return `master` as well. I think that mean we don't need necessarily to distinguish between a regular and a sentinel connection (which would be good, it would mean we don't need to do look at the `workerSettings` constant)

It would mean that non-sentinel connections execute `role` command before handing the connection back, but I think that is okay.